drivers/display/display_led_strip_matrix.c

```diff
@@ -0,0 +1,225 @@
+				  const struct display_buffer_descriptor *desc, const void *buf)
+{
+	const struct led_strip_matrix_config *config = dev->config;
+	struct led_strip_matrix_data *data = dev->data;
+	const uint32_t *u32buf = buf;
+	int rc;
+
+	rc = check_descriptor(config, x, y, desc);
+	if (rc) {
+		LOG_ERR("Invalid descriptor: %d", rc);
+		return rc;
+	}
+
+	for (size_t ypos = y; ypos < (y + desc->height); ypos++) {
+		for (size_t xpos = x; xpos < (x + desc->width); xpos++) {
+			uint32_t color = *u32buf++;
+			size_t idx = pixel_index(config, xpos, ypos);
+			struct led_rgb *pix = &data->pixels[idx];
+
+			pix->r = ((color >> 16) & 0xFF);
+			pix->g = ((color >> 8) & 0xFF);
+			pix->b = ((color) & 0xFF);
+		}
+		u32buf += (desc->pitch - desc->width);
+	}
+
+	rc = led_strip_update_rgb(config->dev, data->pixels, config->width * config->height);
+	if (rc) {
+		LOG_ERR("couldn't update strip: %d", rc);
+	}
+
+	return rc;
+}
+
+static int led_strip_matrix_read(const struct device *dev, const uint16_t x, const uint16_t y,
+				 const struct display_buffer_descriptor *desc, void *buf)
+{
+	const struct led_strip_matrix_config *config = dev->config;
+	struct led_strip_matrix_data *data = dev->data;
+	uint32_t *u32buf = buf;
+
+	rc = check_descriptor(config, x, y, desc);
+	if (rc) {
+		LOG_ERR("Invalid descriptor: %d", rc);
+		return rc;
+	}
+
+	for (size_t ypos = y; ypos < (y + desc->height); ypos++) {
+		for (size_t xpos = x; xpos < (x + desc->width); xpos++) {
+			size_t idx = pixel_index(config, xpos, ypos);
+			struct led_rgb *pix = &data->pixels[idx];
+
+			*u32buf = (0xFF000000 | (pix->r << 16) | (pix->g << 8) | pix->b);
+			u32buf++;
+		}
+		u32buf += (desc->pitch - desc->width);
+	}
+
+	return 0;
+}
+
+static void led_strip_matrix_get_capabilities(const struct device *dev,
+					      struct display_capabilities *capabilities)
+{
+	const struct led_strip_matrix_config *config = dev->config;
+
+	memset(capabilities, 0, sizeof(struct display_capabilities));
+	capabilities->x_resolution = config->width;
+	capabilities->y_resolution = config->height;
+	capabilities->supported_pixel_formats = PIXEL_FORMAT_ARGB_8888;
+	capabilities->current_pixel_format = PIXEL_FORMAT_ARGB_8888;
+	capabilities->screen_info = 0;
+}
+
+static void *led_strip_matrix_get_framebuffer(const struct device *dev)
+{
+	return NULL;
+}
+
+static int led_strip_matrix_blanking_off(const struct device *dev)
+{
+	return -ENOTSUP;
+}
+
+static int led_strip_matrix_blanking_on(const struct device *dev)
+{
+	return -ENOTSUP;
+}
+
+static int led_strip_matrix_set_brightness(const struct device *dev, const uint8_t brightness)
+{
+	return -ENOTSUP;
+}
+
+static int led_strip_matrix_set_contrast(const struct device *dev, const uint8_t contrast)
+{
+	return -ENOTSUP;
+}
+
+static int led_strip_matrix_set_pixel_format(const struct device *dev,
+					     const enum display_pixel_format pixel_format)
+{
+	return -ENOTSUP;
+}
```

Comment:
these all have a NULL check in the API (https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/drivers/display.h#L393), you can drop them all. Some cleanups to do on other drivers as well. :-(